packages/shared/friends/sagas.ts

```diff
@@ -402,6 +416,28 @@ export function getFriends(request: GetFriendsPayload) {
   // TODO: verify if we need to call receivePeerUserData here
 }
 
+export function getFriendRequests(request: GetFriendRequestsPayload) {
+  const friends: FriendsState = getPrivateMessaging(store.getState())
+
+  const fromFriendRequests = friends.fromFriendRequests.slice(
+    request.receivedSkip,
+    request.receivedSkip + request.receivedLimit
+  )
+  const toFriendRequests = friends.toFriendRequests.slice(
+    request.sentSkip,
+    request.sentSkip + request.sentLimit
+  )
+
+  const addFriendRequestsPayload: AddFriendRequestsPayload = {
+    requestedTo: toFriendRequests.map((friend) => friend.userId),
+    requestedFrom: fromFriendRequests.map((friend) => friend.userId),
+    totalReceivedFriendRequests: fromFriendRequests.map((friend) => friend.userId).length,
+    totalSentFriendRequests: toFriendRequests.map((friend) => friend.userId).length
```

Comment:
It's not necessary to do the mapping of both arrays if we are only taking the length (also, both of them were just mapped two lines above).